library/calendar.inc

```diff
@@ -206,4 +209,18 @@ function is_weekend_day($day){
 	}
 }
 
+/**
+ * This function checks if a certain date (YYYY/MM/DD) is a marked as a holiday/closed  event in the events table
+ * @param (int) $day
+ * @return boolean
+ */
+function is_holiday($date){
+	$holidays =array();
+	if(!$holidays_controller) {
+		$holidays_controller = new Holidays_Controller();
+	}
```

Comment:
This function is being called in a loop for each of the days. We don't need to instantiate Holiday_Controller each time.